src/RewardsStreamerMP.sol

```diff
@@ -425,13 +427,14 @@ contract RewardsStreamerMP is
         }
 
         totalRewardsAccrued += newRewards;
-        rewardIndex += (newRewards * SCALE_FACTOR) / totalWeight;
+        rewardIndex += Math.mulDiv(newRewards, SCALE_FACTOR, totalWeight);
         lastRewardTime = block.timestamp < rewardEndTime ? block.timestamp : rewardEndTime;
     }
 
     function _calculateBonusMP(uint256 amount, uint256 lockPeriod) internal pure returns (uint256) {
-        uint256 lockMultiplier = (lockPeriod * MAX_MULTIPLIER * SCALE_FACTOR) / MAX_LOCKUP_PERIOD;
-        return amount * lockMultiplier / SCALE_FACTOR;
+        return Math.mulDiv(
+            Math.mulDiv(amount * lockPeriod, MAX_MULTIPLIER * SCALE_FACTOR, MAX_LOCKUP_PERIOD), 1, SCALE_FACTOR
+        );
```

Comment:
This can be changed to

```
uint256 intermediate = Math.mulDiv(amount * lockPeriod, MAX_MULTIPLIER * SCALE_FACTOR, MAX_LOCKUP_PERIOD);
return intermediate / SCALE_FACTOR;
```